src/ManageCourses.UcasCourseImporter/importer/UcasDataMigrator.cs

```diff
@@ -57,20 +57,24 @@ public void UpdateUcasData()
                 }
             });
 
-            var allProviders = new Dictionary<string, Provider>();
-            MigratePerProvider("upsert providers", inst =>
+            var providerCache = _context.Providers
```

Comment:
How big could this cache potentially get? Will there be enough RAM on the machine?

Reply:
should be fine

Reply:
made a note on the card to check this